commons-vfs2/src/test/java/org/apache/commons/vfs2/provider/http4/Http4ProviderTestCase.java

```diff
@@ -216,4 +219,35 @@ public void testResolveFolderSlashYesRedirectOn() throws FileSystemException {
         testResolveFolderSlash(ConnectionUri + "/read-tests/", true);
     }
 
+    @Test
+    public void testHttp4FileSystemFreeUnusedResources() throws Exception {
+        try (StandardFileSystemManager fileSystemManager = new StandardFileSystemManager()) {
+            fileSystemManager.setConfiguration(StandardFileSystemManager.class.getResource("providers.xml"));
+            // use WeakRef
+            fileSystemManager.setFilesCache(new WeakRefFilesCache());
+            fileSystemManager.init();
+
+            String path = "http4://www.w3schools.com/webservices/tempconvert.asmx?action=WSDL";
```

Comment:
Don't use a remote server. This cause the test to fail randomly and eventually fail all the time. See the other HTTP test in this class which uses an embedded HTTP server.